src/interfaces/IRateLimits.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.8.0;
+
+interface IRateLimits {
+
+    /**********************************************************************************************/
+    /*** Structs                                                                                ***/
+    /**********************************************************************************************/
+
+    /**
+     * @dev Struct representing a rate limit.
+     *      The current rate limit is calculated using the formula:
+     *      `currentRateLimit = min(slope * (block.timestamp - lastUpdated) + lastAmount, maxAmount)`.
+     * @param maxAmount Maximum allowed amount at any time.
+     * @param slope The slope of the rate limit, used to calculate the new limit based on time passed. [tokens / second]
+     * @param lastAmount The amount left available at the last update.
+     * @param lastUpdated The timestamp when the rate limit was last updated.
+     */
+    struct RateLimitData {
+        uint256 maxAmount;
+        uint256 slope;
+        uint256 lastAmount;
+        uint256 lastUpdated;
+    }
+
+    /**********************************************************************************************/
+    /*** Events                                                                                 ***/
+    /**********************************************************************************************/
+
+    /**
+     * @dev Emitted when a rate limit is set.
+     * @param key The identifier for the rate limit.
+     * @param maxAmount The maximum allowed amount for the rate limit.
+     * @param slope The slope value used in the rate limit calculation.
+     * @param lastAmount The amount left available at the last update.
+     * @param lastUpdated The timestamp when the rate limit was last updated.
+     */
+    event RateLimitSet(
```

Comment:
`RateLimitDataSet` since its not the rate limit itself

Reply:
changed